near-sdk-sim/README.md

```diff
@@ -461,3 +453,31 @@ pub fn init () {
     let root = init_simulator(Some(genesis));
 }
 ```
+
+## Loading dumped state from testnet/mainnet contract
+
+This is useful when your test depends on real contract state that exists on testnet and mainnet. For example, storage cost and some gas cost on testnet is depends on the real size of the state that your contract used on testnet and mainnet. So download and start simulator with them gives you a more accurate understanding of gas/storage cost. Another use case is you'd like to migrate your contract state format from old struct defintion to new, in this case fetch testnet states and run integration tests locally will give you confident that your migration works without breaking contract methods.
+
+First you'll need a tool to download state of given contract. Note, the `view_state` rpc doesn't give you the same format that you can passed to simulator as initial state. The major difference is intitial state need every state record typed and has its account id. You can use this [script](https://github.com/near/repro-near-funcall/blob/master/collect-state-records.js) to download multiple contract account state, at given height, in a format that can be used as initial state of near-sdk-sim:
```

Comment:
>You can use this [script]

That's OK as a first-cut, but it's definitely not the experience we want to ship to users. I think we'll need to teach sdk-sim itself to HTTP testnet. This'll help not only this issue, but also with getting realistic costs.

Reply:
I plan to postpone this until we have a plan of make a near-sdk-sim also a cli, after that this script's functionality can be implemented as part of near-sdk-sim-cli